src/uint256.h

```diff
@@ -58,8 +59,20 @@ class base_blob
     friend constexpr bool operator<(const base_blob& a, const base_blob& b) { return a.Compare(b) < 0; }
 
     std::string GetHex() const;
-    void SetHex(const char* psz);
-    void SetHex(const std::string& str);
+    /**
+     * Writes the hex string into the base_blob and returns true iff all of the
+     * checks pass, false otherwise:
+     *   - All characters are hex
+     *   - Input length is WIDTH*2
+     */
+    [[nodiscard]] bool SetHex(std::string_view str)
+    {
+        if (WIDTH * 2 != str.size() || !IsHex(str)) return false;
+        SetHexFragile(str);
+        return true;
+    }
+    /** Like SetHex, but accepting any invalid input, thus fragile and deprecated */
+    void SetHexFragile(std::string_view str);
```

Comment:
`-Fragile` feels slightly off to me as well.
`-Unchecked` seems to imply that the function itself may crash.
`-Lenient` feels closer to the truth but sounds safe to use.

I recently came across `MakeRandDeterministicDANGEROUS()`, maybe something like that suffix would work?

Reply:
It is dangerous because input isn't length-checked, meaning extra hex chars may be ignored, and an accidentally short copy-paste of hex chars will be silently padded with zeroes.

> Maybe just ...Deprecated()?

As in https://github.com/bitcoin/bitcoin/pull/30482?notification_referrer_id=NT_kwDOCkdNarUxMTU3NjYyODE4MzoxNzI0NDUwMzQ#discussion_r1685726873, I wonder what the policy around deprecation is. Are there prior examples of deprecation on a non-RPC level you can point to?

Reply:
But production code calls it indirectly, so it's not exactly deprecated, just direct access is discouraged, right?

Reply:
I'm fine with that if you are

Reply:
> Only src/util/time.h, but those are just type-related; And src/logging.h, but those are just aliases; So I don't think they apply here.

Thanks, those are good examples. I guess they are/were used more widely and directly compared to SetHex(), making deprecation a more obvious approach. If you deem this sufficiently big to break up into multiple PRs, I trust your judgement.

`SetHexDeprecated` or `SetHexDiscouraged` work for me.